megalodon/src/firefish/api_client.ts

```diff
@@ -284,25 +287,34 @@ namespace FirefishAPI {
         application: null,
         language: null,
         pinned: null,
-        emoji_reactions: typeof n.reactions === 'object' ? mapReactions(n.reactions, n.myReaction) : [],
+        // Use emojis list to provide URLs for emoji reactions.
+				emoji_reactions: mapReactions(n.emojis!, n.reactions, n.myReaction),
         bookmarked: false,
         quote: n.renote !== undefined && n.text !== null
       }
     }
 
-    export const mapReactions = (r: { [key: string]: number }, myReaction?: string | null): Array<MegalodonEntity.Reaction> => {
+    export const mapReactions = (emojis: Array<FirefishEntity.Emoji>, r: { [key: string]: number }, myReaction?: string | null): Array<MegalodonEntity.Reaction> => {
+      // Map of emoji shortcodes to image URLs.
+			const emojiUrls = new Map<string, string>(
+				emojis.map((e) => [e.name, e.url]),
+			);
       return Object.keys(r).map(key => {
-        if (myReaction && key === myReaction) {
-          return {
-            count: r[key],
-            me: true,
-            name: key
-          }
-        }
+        // Strip colons from custom emoji reaction names to match emoji shortcodes.
+				const shortcode = key.replace(/:/g, "");
+				// If this is a custom emoji (vs. a Unicode emoji), find its image URL.
+				const url = emojiUrls.get(shortcode);
+				// Finally, remove trailing @. from local custom emoji reaction names.
+				const name = shortcode.replace("@.", "");
```

Comment:
Just confirm.
This logic means you want to remove `:` and @.` from emoji_reactions, right?
For example,
`{ count: 1, me: false, name: ':ablobfoxexplode@.:' }` -> `{ count: 1, me: false, name: 'ablobfoxexplode' }`

Reply:
Yes

Reply:
Thanks, I got it.